lib/internal/Magento/Framework/Console/Cli.php

```diff
@@ -256,4 +254,23 @@ private function documentRootResolver(array $config = []): array
 
         return array_merge_recursive($config, $params);
     }
+
+    /**
+     * Generate and return the Command Loader
+     *
+     * @throws \LogicException
+     * @throws \BadMethodCallException
+     */
+    private function getCommandLoader(): Console\CommandLoader\CommandLoaderInterface
+    {
+        $commandLoaders = [];
+        if (class_exists(SetupCommandLoader::class)) {
```

Comment:
Could you explain why this class might not exist?

Reply:
My goal was to avoid regressions, so I tried to translate this class literally to mirror how `CommandList` classes were used. Here's what inspired the check: https://github.com/magento/magento2/blob/1e846a4aa38f1ff5d7238d70df9d887b3bdf5acb/lib/internal/Magento/Framework/Console/Cli.php#L146

I'm not sure if any likely scenario exists where the `Magento\Setup` classes aren't available, but that's presumably the hedge this check is betting against. Let me know how you'd like to move forward... I'm happy to remove the check if we don't need to worry about cases where users of Magento have excluded "setup" classes from autoloading.